Assets/HoloToolkit/Input/Scripts/KeywordManager.cs

```diff
@@ -2,79 +2,97 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System.Collections.Generic;
-using System.Linq;
 using UnityEngine;
-using UnityEngine.Events;
 using UnityEngine.Windows.Speech;
 
 namespace HoloToolkit.Unity
 {
     /// <summary>
-    /// KeywordManager allows you to specify keywords and methods in the Unity
+    /// KeywordManager allows you to specify keywords in the Unity
     /// Inspector, instead of registering them explicitly in code.
     /// This also includes a setting to either automatically start the
     /// keyword recognizer or allow your code to start it.
     ///
     /// IMPORTANT: Please make sure to add the microphone capability in your app, in Unity under
     /// Edit -> Project Settings -> Player -> Settings for Windows Store -> Publishing Settings -> Capabilities
     /// or in your Visual Studio Package.appxmanifest capabilities.
+    /// 
+    /// Please, also make sure to also set your keyword responses in the Inspector window under Keywords and Responses Array.
     /// </summary>
-    public partial class KeywordManager : MonoBehaviour
+    public partial class KeywordManager : Singleton<KeywordManager>
```

Comment:
The KeywordRecognizer has been written in a way to optimize turning them on and off when their keywords are actually needed. Leaving all keywords on and ignoring their events if the state isn't correct creates more work for the recognition engine (more keywords to try to differentiate and recognize against, increasing the chance of a false accept or reject) and is less efficient overall. The old model worked this way, since adding and removing keywords used to be expensive, but we worked with Unity to improve the implementation into what exists today.